### PR TITLE
fix(editor): Add sticky to bottom editor when out of view

### PR DESCRIPTION
**Pull request by meel-hd**
## Proposed changes
The logic used to the calculate scroll offset based on the closest tbody tag which is fine unless the space between them is very large (like in issue #3507) which hides the translation input. As a fix we calculate based on a closer element that can't grow in height very much to hide the translation input.

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

### CODE REVIEW

**Comment by nijel** [comment]
The main intention here is to show the source string while the user navigates using the tab key. Your change destroys this (and mostly reverts to browser default behavior).

The whole thing is problematic because the content we want to show doesn't have to fit on the screen.

Ideas which could make it behave nicer:

* Move the viewport in a way that the editor is at the bottom, showing as much content above as possible.
* Do not adjust the viewport when the current editor was already focused.

**Comment by meel-hd** [comment]
Or the editor should stick in the bottom of the screen when it goes out of screen. It will insure the user can scroll freely to view anything and be able to edit anywhere
**Example**

https://github.com/user-attachments/assets/56695901-cfb4-4e50-a53a-3594899b781a

**Comment by nijel** [comment]
Great idea, it definitely makes sense as long as the editor is focused.

File: weblate/templates/zen-units.html [hunk 1]
```diff
@@ -88,7 +88,10 @@
 </td>
 {% endif %}
 <td {% if user.profile.zen_mode != user.profile.ZEN_HORIZONTAL %}colspan="2"{% endif %} class="translator">
-{% crispy item.form %}
+  <button id="hide-sticky" class="btn btn-light btn-sm" title="{% trans "Collapse" %}" >
```
Comment (nijel) [inline weblate/templates/zen-units.html:91]: This form is there for each string, so using `id` is probably not a good idea as there will be more of them in the page.

Comment (meel-hd) [reply weblate/templates/zen-units.html:91]: Yeah that's not a problem in this case, we select only the closest hide btn to the string editor we are in. So others are ignored

Comment (nijel) [reply weblate/templates/zen-units.html:91]: Isn't `id` supposed to be unique in the document?

Comment (meel-hd) [reply weblate/templates/zen-units.html:91]: Yes that's what is said. But to be honest I'm not sure if it is that important or enforced by any means.

**Pull request merged**